cfbs/commands.py

```diff
@@ -453,88 +458,181 @@ def _someone_needs_me(this) -> bool:
+        )
+
+        if input_def["type"] == "list":
+            def_subtype = input_def["subtype"]
+            data_subtype = input_data["subtype"]
+            if type(def_subtype) != type(data_subtype):
+                log.warning(
+                    "Failed to update input data for module '%s': " % module_name
+                    + "Different subtypes in list ('%s' != '%s')."
+                    % (type(def_subtype).__name__, type(data_subtype).__name__)
+                )
+                raise InputDataUpdateFailedException()
+            if isinstance(def_subtype, list):
+                if len(def_subtype) != len(data_subtype):
+                    log.warning(
+                        "Failed to update input data for module '%s': " % module_name
+                        + "Different amount of elements in list ('%s' != '%s')."
+                        % (len(def_subtype), len(data_subtype))
+                    )
+                    raise InputDataUpdateFailedException()
+                for i in range(len(def_subtype)):
+                    _check_keys(def_subtype[i], data_subtype[i], ("key", "type"))
+                    changes_made |= _update_keys(
+                        def_subtype[i],
+                        data_subtype[i],
+                        ("label", "question", "default"),
+                    )
+            elif isinstance(def_subtype, dict):
+                _check_keys(def_subtype, data_subtype, ("type",))
+                changes_made |= _update_keys(
+                    def_subtype, data_subtype, ("label", "question", "default")
+                )
+            else:
+                user_error(
+                    "Unsupported subtype '%s' in input definition for module '%s'."
+                    % (type(def_subtype).__name__, module_name)
+                )
+        return changes_made
+
+    changes_made = False
+    for i in range(len(input_def)):
+        changes_made |= _update_variable(input_def[i], input_data[i])
+    return changes_made
+
+
 @commit_after_command("Updated module%s", [PLURAL_S])
 def update_command(to_update):
     config = CFBSConfig.get_instance()
-    index = config.index
+    build = config["build"]
 
-    if to_update:
-        to_update = [Module(m) for m in to_update]
-        index.translate_aliases(to_update)
-        skip = (
-            m for m in to_update if all(n["name"] != m.name for n in config["build"])
-        )
-        for m in skip:
-            log.warning("Module '%s' not in build. Skipping its update.", m.name)
-            to_update.remove(m)
-    else:
-        # Update all modules in build if no modules are specified
-        to_update = [Module(m["name"]) for m in config["build"]]
+    # Update all modules in build if none specified
+    to_update = (
+        [Module(m) for m in to_update]
+        if to_update
+        else [Module(m["name"]) for m in build]
+    )
```

Comment:
This part (and many changes below) doesn't seem to be related to handling inputs as part of `update`. Ideally, it should be in a separate commit, but not a big deal if it would be hard to separate.